reactor-netty-http/src/test/java/reactor/netty/http/client/HttpClientTest.java

```diff
@@ -614,6 +618,97 @@ void sslExchangeRelativeGet() throws SSLException {
 		assertThat(responseString).isEqualTo("hello /foo");
 	}
 
+	@Test
+	void expectedConnectionPoolsEnabled() throws SSLException {
+		ArgumentCaptor<String> argumentCaptor = ArgumentCaptor.forClass(String.class);
+		Logger spyLogger = Mockito.spy(log);
+		Loggers.useCustomLoggers(s -> spyLogger);
+
+		SslContext sslServer = SslContextBuilder.forServer(ssc.certificate(), ssc.privateKey())
+				.build();
+
+
+		disposableServer =
+				createServer()
+						.secure(ssl -> ssl.sslContext(sslServer))
+						.handle((req, resp) -> resp.sendString(Flux.just("hello ", req.uri())))
+						.bindNow();
+
+		ConnectionProvider connectionProvider = ConnectionProvider.builder("expected-connection-pool").expectedConnectionPools(1).build();
```

Comment:
IMO it is better to call it maximum connection pools. Wdyt?

```suggestion
		ConnectionProvider connectionProvider = ConnectionProvider.builder("max-connection-pools").maxConnectionPools(1).build();
```